indico/web/flask/session.py

```diff
@@ -208,18 +233,17 @@ def save_session(self, app, session, response):
             # If the session has not been modified we only store if it needs to be refreshed
             return
 
-        if config.SESSION_LIFETIME > 0:
+        if config.SESSION_LIFETIME > 0 or session.hard_expiry:
             # Setting session.permanent marks the session as modified so we only set it when we
             # are saving the session anyway!
             session.permanent = True
 
         storage_ttl = self.get_storage_lifetime(app, session)
-        cookie_lifetime = self.get_expiration_time(app, session)
-        session['_expires'] = datetime.now() + storage_ttl
-
-        if refresh_sid:
-            self.storage.delete(session.sid)
-            session.sid = self.generate_sid()
```

Comment:
Why did you remove this? It's a security issue ("session fixation") to not reset the session ID when logging in/logging out

Reply:
This was a mistake. Mea culpa.